### PR TITLE
curvefs/tool: fix error

### LINKED ISSUE #932: curvefs_tool createfs have no retry when timeout
**Describe the bug (描述bug)**
截屏2022-01-05 上午11 20 05

**To Reproduce (复现方法)**
Use curvefs_tool to create fs 'curvefs_tool create-fs -fsName=file'

**Expected behavior (期望行为)**

**Versions (各种版本)**
OS:
Compiler:
curve-mds:
curve-chunkserver:
curve-snapshotcloneserver:
curve-sdk:
nebd:
curve-nbd:

**Additional context/screenshots (更多上下文/截图)**

### PR DESCRIPTION
**Pull request by Cyber-SiKu**
### What problem does this PR solve?

Issue Number: close #932 #922 #945 

Problem Summary:

1. fix some error in  create-fs
2. fix some error output in curvefs_tool
3. adjust output of umount-fs
4. adjust output of create-fs: add fs exist (create same fs name but the
parameters are inconsistent)
5. adjust output of query-copyset: indicate that the number of statuses
is inconsistent with the peers in the info
6. adjust output of query-fs: add fs not found
7. adjust output of delete-fs: add fs not found
8. fix check-copyset
9. fix query-copyset fail when query same copyset: add requset
copysetkey when sure

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curvefs/src/tools/copyset/curvefs_copyset_base_tool.h [hunk 1]
```diff
@@ -31,13 +31,13 @@
...
 #include "curvefs/src/mds/topology/deal_peerid.h"
-#include "curvefs/src/tools/check/curvefs_copyset_check.h"
+#include "curvefs/src/tools/copyset/curvefs_copyset_status.h"
 
 namespace curvefs {
 namespace tools {
 namespace copyset {
 
-uint64_t GetCopysetKey(uint64_t copysetId, uint64_t poolId);
+uint64_t GetCopysetKey(const uint64_t& copysetId, const uint64_t& poolId);
```
Comment (SeanHai) [inline curvefs/src/tools/copyset/curvefs_copyset_base_tool.h:40]: GetCopysetKey(uint64_t copysetId, uint64_t poolId) is enough.

File: curvefs/src/tools/umount/curvefs_umount_fs_tool.cpp [hunk 2]
```diff
@@ -98,14 +98,26 @@ bool UmountFsTool::AfterSendRequestToHost(const std::string& host) {
                      << " failed, errorcode= " << controller_->ErrorCode()
                      << ", error text " << controller_->ErrorText()
                      << std::endl;
-    } else if (response_->statuscode() != curvefs::mds::FSStatusCode::OK) {
+    } else if (response_->statuscode() == curvefs::mds::FSStatusCode::OK) {
```
[the remark this answers is not in the document]
Comment (Cyber-SiKu) [reply curvefs/src/tools/umount/curvefs_umount_fs_tool.cpp:101]: > Use 'switch case' is better?

fix

File: curvefs/src/tools/query/curvefs_fs_query.cpp [hunk 3]
```diff
@@ -94,10 +94,32 @@ bool FsQueryTool::AfterSendRequestToHost(const std::string& host) {
                      << ", error text " << controller_->ErrorText() << "\n";
         return false;
     } else if (show_) {
-        std::cout << response_->DebugString() << std::endl;
+        if (response_->statuscode() == mds::FSStatusCode::OK) {
+            std::cout << response_->DebugString() << std::endl;
```
[the remark this answers is not in the document]
Comment (Cyber-SiKu) [reply curvefs/src/tools/query/curvefs_fs_query.cpp:98]: > Does log is necessary when success. The log will mixed with query result.

fix

File: curvefs/src/tools/copyset/curvefs_copyset_base_tool.h (hunk 1 above)
Comment (Cyber-SiKu) [reply curvefs/src/tools/copyset/curvefs_copyset_base_tool.h:40]: > GetCopysetKey(uint64_t copysetId, uint64_t poolId) is enough.

fix

File: curvefs/src/tools/check/curvefs_copyset_check.cpp [hunk 4]
```diff
@@ -22,86 +22,47 @@
...
-        key_.push_back((poolId << 32) | copysetId);
-    }
-    AddRequest(request);
-
-    service_stub_func_ = std::bind(
-        &curvefs::metaserver::copyset::CopysetService_Stub::GetCopysetsStatus,
-        service_stub_.get(), std::placeholders::_1, std::placeholders::_2,
-        std::placeholders::_3, nullptr);
+    FLAGS_detail = true;  // try to get copyset status
+    queryCopysetTool_ = std::make_shared<query::CopysetQueryTool>("", false);
```
Comment (wu-hanqing) [inline curvefs/src/tools/check/curvefs_copyset_check.cpp:46]: the first parameter of CopysetQueryTool's constructor is `cmd`, so, what's the meaning of `""`?

File: curvefs/src/tools/copyset/curvefs_copyset_base_tool.cpp [hunk 5]
```diff
@@ -32,7 +32,7 @@ namespace curvefs {
 namespace tools {
 namespace copyset {
 
-uint64_t GetCopysetKey(uint64_t copysetId, uint64_t poolId) {
+uint64_t GetCopysetKey(const uint64_t& copysetId, const uint64_t& poolId) {
```
[the remark this answers is not in the document]
Comment (Cyber-SiKu) [reply curvefs/src/tools/copyset/curvefs_copyset_base_tool.cpp]: > for built-in type, pass by value is better

has fixed

File: curvefs/src/tools/check/curvefs_copyset_check.cpp (hunk 4 above)
Comment (Cyber-SiKu) [reply curvefs/src/tools/check/curvefs_copyset_check.cpp:46]: > the first parameter of CopysetQueryTool's constructor is `cmd`, so, what's the meaning of `""`?

it was called by another CurvefsTool, the value cmd is useless, so set "".

**Review by SeanHai — APPROVED** [review]

**Review by wu-hanqing — APPROVED** [review]

**Pull request merged**